recipe/meta.yaml

```diff
@@ -32,7 +32,7 @@ requirements:
     - python
     - pip
     - setuptools
-    - pybind11 >=2.8,<2.9
+    - pybind11 >=2.8
```

Comment:
Nice catch, and justifies your comments about the harm of version pinning

Reply:
Hm yes... To be honest, I don't see things quite as religiously as others (e.g. I can understand why someone releasing wheels might want to be conservative for ABI-breaking packages), but it's at odds with the packaging mantra of reflecting upstream dependency constraints (because there might be arbitrarily many reasons for those).

In such cases, the packaging takes some extra care & reflection, but since we have the machinery to track things correctly (and run the full test suite), I think this is not too radical of a departure.